common/persistence/dataInterfaces.go

```diff
@@ -574,22 +574,22 @@ type (
 		FinishedExecutionTTL int32
 
 		// Mutable state
-		UpsertActivityInfos           []*ActivityInfo
-		DeleteActivityInfos           []int64
-		UpserTimerInfos               []*TimerInfo
-		DeleteTimerInfos              []string
-		UpsertChildExecutionInfos     []*ChildExecutionInfo
-		DeleteChildExecutionInfo      *int64
-		UpsertRequestCancelInfos      []*RequestCancelInfo
-		DeleteRequestCancelInfo       *int64
-		UpsertSignalInfos             []*SignalInfo
-		DeleteSignalInfo              *int64
-		UpsertSignalRequestedIDs      []string
-		DeleteSignalRequestedID       string
-		NewBufferedEvents             *SerializedHistoryEventBatch
-		ClearBufferedEvents           bool
-		NewBufferedReplicationTask    *BufferedReplicationTask
-		DeleteBufferedReplicationTask *int64
+		UpsertActivityInfos            []*ActivityInfo
+		DeleteActivityInfos            []int64
+		UpserTimerInfos                []*TimerInfo
+		DeleteTimerInfos               []string
+		UpsertChildExecutionInfos      []*ChildExecutionInfo
+		DeleteChildExecutionInfo       *int64
+		UpsertRequestCancelInfos       []*RequestCancelInfo
+		DeleteRequestCancelInfo        *int64
+		UpsertSignalInfos              []*SignalInfo
+		DeleteSignalInfo               *int64
+		UpsertSignalRequestedIDs       []string
+		DeleteSignalRequestedID        string
+		NewBufferedEvents              *SerializedHistoryEventBatch
+		ClearBufferedEvents            bool
+		NewBufferedReplicationTask     *BufferedReplicationTask
+		DeleteBufferedReplicationTasks []int64
```

Comment:
I thought we only process single buffered event in each transaction.  What is the use case for processing more than one buffered replication task?

Reply:
https://github.com/uber/cadence/blob/master/service/history/historyReplicator.go#L419